src/main/java/io/vertx/redis/client/impl/RedisClusterClient.java

```diff
@@ -139,72 +142,47 @@ public RedisClusterClient(Vertx vertx, NetClientOptions tcpOptions, PoolOptions
   @Override
   public Future<RedisConnection> connect() {
     final Promise<RedisConnection> promise = vertx.promise();
-    // attempt to load the slots from the first good endpoint
-    connect(connectOptions.getEndpoints(), 0, promise);
+    getSlots()
```

Comment:
I think it can be replaced by getSlots().compose(slots -> ...) instead of onSuccess/onFailure

Reply:
OK, I see how all the comments when put together make the code quite a bit nicer. Will do!

Reply:
Actually no, we do need the extra `Promise` to pass to the `connect()` method, because that doesn't return a `Future`. It has a callback parameter, because it may retry and passing around the callback is cheaper than creating a `Future` for each retry attempt.

If we want to keep that, future combinators won't work. In which case, I think `onSuccess()` / `onFailure()` on the `Future` returned by `getSlots()` are just overall better.

Reply:
For now, I at least pushed a commit that makes `getSlots()` use a proper context-bound promise.